x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/ml/QueryPage.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.protocol.xpack.ml;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Generic wrapper class for a page of query results and the total number of
+ * query results.<br>
+ * {@linkplain #count()} is the total number of results but that value may
+ * not be equal to the actual length of the {@linkplain #results()} list if from
+ * &amp; take or some cursor was used in the database query.
+ */
+public final class QueryPage<T extends ToXContent> implements ToXContentObject {
```

Comment:
I think we do not need this class in the client. In the server side, it saves us from duplicating the structure `count, List<T>` from a number of responses. But on the client side, it doesn't seem like it's saving us a lot. I think it would be simpler to have the count and the List<Job> in the `GetJobResponse` directly. I've been doing the same for the get buckets API. Let me know what you think.